vhpidirect/quickstart/logicvector/main.c

```diff
@@ -0,0 +1,33 @@
+#include <stdio.h>
+static const char HDL_LOGIC_STATE[] = { 'U', 'X', '0', '1', 'Z', 'W', 'L', 'H', '-'};
+
+char logic_vec[9];
+int logic_vec_size = 9;
+
+int getLogicVecSize(){
+  return logic_vec_size;
+}
+
+char* getLogicVec(){
+  printf("1D Array Logic Values [%d]:\n", logic_vec_size);
+  for(int i = 0; i < logic_vec_size; i++){
+    logic_vec[i] = i;
+    printf("[%d] = %c\t", i, HDL_LOGIC_STATE[i]);
+  }
+  printf("\n");
+  return logic_vec;
+}
+
+/* char* getULogicMat(){
```

Comment:
It seems that you didn't include this in the end because, as you found out, sharing multidimensional arrays does require `ghdl.h` in practice. Let's just remove this comment.

Reply:
I believe I could get a constrained 2D array to work without ghdl.h.
Will try in a commit before removing.

I just thought that 2D may be out of scope for the quickstart is all.

Reply:
Well, I think that quickstart is about being quick and not too large, rather than being extremely easy/simple. We are already adding very simple examples, so we can add some intermediate ones before going to the full/complex solution. Hence, it's ok if you want to include here, or if you want to create a separate constrained 2D example.

Actually, as a user of VUnit, there is this `integer_array_t` type with a `load_csv` function: https://github.com/VUnit/vunit/blob/master/examples/vhdl/array_axis_vcs/src/test/tb_axis_loop.vhd#L50. That's quite useful to export matrices from Matlab/Octave to CSV files and then read them as a 2D array in VHDL. However, it is limited to integers. Hence, floats/singles/doubles need to be converted to fixed-point, then scaled and scaled down (or treated as `std_logic_vector`/`signed`/`unsigned`) in the testbench. Using VHPIDIRECT is a nice alternative since it allows to pass a pointer to a 2D array of doubles and use them in VHDL as reals (https://ghdl.github.io/ghdl-cosim/vhpidirect/declarations.html#restrictions-on-type-declarations). Hence, it would be useful to have a quickstart example of a constrained 2D array of floating point numbers.

Reply:
I see the commented out code was for an unbounded array. Yeah, that'd be in need of ghdl.h. I have a commit with a bounded 2D working.
Could make VecB 2D if we wanted to show off within the quickstart. I'm easy on that, may just add some excitment to the intro 😄

Reply:
Just keep A and B in here, move C to a separate example (quickstart/matrices) and I will copy/extend it to floats. Does it sound good?

Reply:
Yeah, sure. Will clean it now.